OBAKitCore/Network/RESTAPIService/RESTAPIService+Processing.swift

```diff
@@ -0,0 +1,93 @@
+//
+//  RESTAPIService+Processing.swift
+//  OBAKitCore
+//
+//  Created by Alan Chu on 12/28/22.
+//
+
+import Foundation
+
+extension RESTAPIService {
+    nonisolated func getData(for url: URL) async throws -> (Data, HTTPURLResponse) {
+        var request = URLRequest(url: url, cachePolicy: .useProtocolCachePolicy, timeoutInterval: 10.0)
+        request.setValue("gzip", forHTTPHeaderField: "Accept-Encoding")
+
+        let (data, response): (Data, URLResponse)
+        do {
+            logger.info("Begin network request for \(url, privacy: .public)")
+            (data, response) = try await dataLoader.data(for: request)
+            logger.info("Finish network request for \(url, privacy: .public)")
+        } catch (let error as NSError) {
+            logger.error("Failed network request for \(url, privacy: .public): \(error, privacy: .public)")
+            if errorLooksLikeCaptivePortal(error) {
+                throw APIError.captivePortal
+            } else {
+                throw error
+            }
+        }
+
+        guard let httpResponse = response as? HTTPURLResponse else {
+            logger.error("Failed network request for \(url, privacy: .public): missing response.")
+            throw APIError.networkFailure(nil)
+        }
+
+        guard 200...299 ~= httpResponse.statusCode else {
+            if httpResponse.statusCode == 404 {
+                logger.error("Failed network request for \(url, privacy: .public): 404 not found.")
+                throw APIError.requestNotFound(httpResponse)
+            } else {
+                logger.error("Failed network request for \(url, privacy: .public): \(httpResponse).")
+                throw APIError.requestFailure(httpResponse)
+            }
+        }
+
+        // The REST API doesn't do a good job of surfacing what should be 404 errors.
+        // If you request a valid endpoint, but provide it with a bogus piece of
+        // data (e.g. a non-existent Stop ID), it should return a 404 error to you.
+        // Instead, it gives a 200 and a blank body.
+        if httpResponse.expectedContentLength == 0 && httpResponse.statusCode == 200 {
+            logger.error("Failed network request for \(url, privacy: .public): 404 not found.")
+            throw APIError.requestNotFound(httpResponse)
+        }
```

Comment:
Continuing previous discussion: I kept the handling of `empty 200` around. The difference now is both `statusCode == 404` or `response.isEmpty AND statusCode == 200` returns `APIError.requestNotFound`